src/main/java/core/basesyntax/Palindrome.java

```diff
@@ -21,6 +21,11 @@
      * <p>Результат: true</p>
      */
     public boolean isPalindrome(String text) {
-        return false;
+        String original = text.replaceAll("[^a-zA-Z0-9_]", "")
+                .replaceAll(" ", "").toLowerCase();
+        StringBuilder firstPart =
+                new StringBuilder(original.substring(0, original.length() / 2));
```

Comment:
do we need to divide our string? Let's replace all string and compare it with input data

Reply:
Do I get your point right?
You want me to create a StringBuilder type of object named, for example, formattedText,
then to replace all the signs we need to remove with a "" regex,
then in return statement to compare first part of formattedText with the second one?

Reply:
> do we need to divide our string? Let's replace all string and compare it with input data

Do I get your point right?
You want me to create a StringBuilder type of object named, for example, formattedText,
then to replace all the signs we need to remove with a "" regex,
then in return statement to compare first part of formattedText with the second one?

Reply:
Or should I create to StringBuilder objects like this:

```
StringBuilder formattedText = new StringBuilder(text.replaceAll("[^a-zA-Z0-9_-]", "")
                .replaceAll(" ", "").toLowerCase());
        System.out.println(formattedText);

        StringBuilder palindromeNew =
                new StringBuilder(formattedText.replace((formattedText.length() / 2), formattedText.length(), "")
                .reverse());
        
        return formattedText.equals(palindromeNew.append(palindromeNew));
```

But then I need to use a ternary operator to add a middle char if exists...
Or create a boolean type variable. Won't it be an overloading of the code?

Reply:
OK, I got it... :D